controller/index.js

```diff
@@ -1,338 +1,316 @@
-            if (err){
-              res.send(err, 500);
-            } else {
-              // send back image
-              res.sendFile( file );
-            }
-          });
-          //res.json( data );
-       } else {
-         controller.Error(req, res);
-       }
-    };
-  
-  
-    var fileName = Github.thumbnailExists(key, req.query);
-    if ( fileName ){
-      res.sendFile( fileName );
+    var key = ['github', req.params.user, req.params.repo, req.params.file].join(':')
+    var dir = Github.cacheDir() + '/thumbs/'
+    req.query.width = parseInt(req.query.width, 10) || 150
+    req.query.height = parseInt(req.query.height, 10) || 150
+    req.query.f_base = dir + key + '/' + req.query.width + '::' + req.query.height
+
+    var _reply = function (err, data) {
+      if (err) return res.json(err, 500)
+      if (!data) controller.Error(req, res)
+
+      // generate a thumbnail
+      Github.generateThumbnail(data[0], key, req.query, function (err, file) {
+        if (err) {
+          res.status(500).send(err)
+        } else {
+          // send back image
+          res.sendFile(file)
+        }
+      })
+    }
+
+    var fileName = Github.thumbnailExists(key, req.query)
+    if (fileName) {
+      res.sendFile(fileName)
     } else {
-  
-     if ( req.params.user && req.params.repo && req.params.file ){
-        req.params.file = req.params.file.replace('.geojson', '');
-        Github.find(req.params.user, req.params.repo, req.params.file, req.query, _reply );
-      } else if ( req.params.user && req.params.repo, req.query ) {
-        Github.find(req.params.user, req.params.repo, null, req.query, _reply );
+      if (req.params.user && req.params.repo && req.params.file) {
+        req.params.file = req.params.file.replace('.geojson', '')
+        Github.find(req.params.user, req.params.repo, req.params.file, req.query, _reply)
+      } else if (req.params.user && req.params.repo && req.query) {
+        Github.find(req.params.user, req.params.repo, null, req.query, _reply)
       } else {
-        controller.notFound(req, res);
+        controller.notFound(req, res)
       }
     }
-  };
-  
-  // 
-  controller.getRepo = function(req, res){
-      var key = ['github'];
-  
-      // method to respond to model finds
-      var _send = function( err, data ){
-        if ( err ){
-          res.json( err, 500 );
+  }
+
+  controller.getRepo = function (req, res) {
+    // method to respond to model finds
+    var _send = function (err, data) {
+      if (err) return res.json(err, 500)
+      if (!data) return controller.Error(req, res)
+
+      var len = data.length
+      var allTopojson = []
+
+      var processTopojson = function (topology) {
+        allTopojson.push(topology)
+        if (allTopojson.length === len) {
+          res.json(allTopojson[0])
+        }
+      }
+
+      if (req.query.topojson) {
+        data.forEach(function (d) {
+          Topojson.convert(d, function (err, topology) {
```

Comment:
@chelm `Topojson` isn't actually defined anywhere in this file. Any idea where it came from?